### PR TITLE
Support vendored libs in dependencies

### LINKED ISSUE #110: not mangling not working
What am I doing wrong here? 

``` bash
$ pod package --no-mangle Mapbox-iOS-SDK
[!] podspec has binary-only depedencies, mangling not possible.
```

I've tried `--library` and `--embedded` as well, but no change.

### PR DESCRIPTION
**Pull request by jasl8r**
This adds a few features:
- Allows binary dependencies when not mangling
- Includes vendored libraries and frameworks in final binary output
- Adds prelink option to merge all symbols into a single object and strip unused code
- Adds exported symbols option to limit exposed symbols, providing functionality similar to mangling

Resolves #110 #71 #26 and maybe #70

### CODE REVIEW

File: lib/cocoapods-packager/pod_utils.rb [hunk 1]
```diff
@@ -38,6 +51,37 @@ def install_pod(platform_name, sandbox)
...
+      end
+
+      def static_libraries(installer, root)
+        installer.pod_targets.map do |target|
+          next if target.product_module_name == @spec.name
+
+          if target.should_build?
+            File.join(root, "lib#{target.product_module_name}.a")
+          end
+        end
```
Comment (segiddins) [inline lib/cocoapods-packager/pod_utils.rb]: `.compact` ?

Comment (jasl8r) [inline lib/cocoapods-packager/pod_utils.rb]: To remove redundant dependencies I assume?  In case multiple deps have the same binary deps?

Comment (segiddins) [inline lib/cocoapods-packager/pod_utils.rb]: no, because this block can return `nil`

**Comment by neonichu** [comment]
@jasl8r Thanks for the PR! :tada: 

Could you make the change Samuel suggested and rebase?

**Comment by jasl8r** [comment]
Sure will do.

**Comment by ExoticObjects** [comment]
This pull request would add some nice functionality - any idea when it can actually happen?

**Comment by jasl8r** [comment]
Ah, sorry, I really dropped the ball.  I'm going to look at this now.

**Comment by jasl8r** [comment]
Rebased on master and added the `compact` methods.

**Comment by santialvarezcolombo** [comment]
Trying out this changes, I found that when packaging from a podspec with a valid **default_subspecs** field, the only headers that get downloaded are the ones in that default subspec. Even without that field in the podspec, headers for dependencies don't get downloaded either. 

As an example, I used [Google's podspec](https://github.com/CocoaPods/Specs/blob/master/Specs/Google/1.3.2/Google.podspec.json) with the packager, and trying to use Analytics in a project, I get the error **'GAI.h' file not found** which is reasonable, since it's a header file needed by **Analytics.h** and its not being downloaded to the folder **Headers/Google** created by the packager.

If I use a Podfile containing the Google SDK as stated in the [CocoaPods page](https://cocoapods.org/?q=google), for the same example as before, I get that missing header in the following path *Pods/GoogleAnalytics/Headers/Public/GAI.h* so the project compiles with no errors.

Am I doing something wrong? Or this is actually a problem of the tool/this PR?

I'm adding this here, since Google uses binary dependencies, and this example is impossible to run without the changes of this PR. If it doesn't belong here, I can add it as an issue to the repo.

**Comment by danielbowden** [comment]
This looks really good. Has it stalled or can it be merged? I'd be keen to try it out.

**Comment by neonichu** [comment]
It conflicts now, but I'd be happy to merge if it is rebased or I'm also happy to pick it up and get merged.

**Comment by xquezme** [comment]
Any updates of this?

**Comment by norlin** [comment]
Is this issue is still alive? I've faced with a problem – I have one pod with a vendored framework (a static library), and I need to use this pod as a dependency in another pod (swift dynamic framework).

Currently I get "pod target has transitive dependencies that include static binaries" error when doing `pod install`.

Is it related to this issue?

**Comment by EuanChan** [comment]
@norlin I'm facing the same problem, have you found a solution? 
I've try some other pr, but not one really fix the problem.

**Comment by norlin** [comment]
@EuanChan not really, I just rewrite the static library to a framework, which is work great in the same setup.

**Comment by amorde** [comment]
Going to close as this has become outdated, and I believe this functionality was added 👍

**Pull request closed without merging**